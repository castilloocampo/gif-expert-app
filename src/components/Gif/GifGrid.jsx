import React, { useEffect, useState } from 'react';
import PropTypes from 'prop-types';
import GifGridItem from './GifGridItem';

const API_KEY = "t8efOi4sM27mFICI9gxlJQHSrdnBulnH";



const GifGrid = ({ category }) => {

  const [images, setImages] = useState([]);

  const getGifs = async () => {
    const url = `https://api.giphy.com/v1/gifs/search?q=${category}&limit=10&api_key=${API_KEY}`;
    const response = await fetch(url);
    const { data } = await response.json();
    const gifs = data.map(img => {
      return {
        id: img.id,
        title: img.title,
        url: img.images?.downsized_medium.url
      }
    })
    setImages(gifs);
  }

  useEffect(() => {
    getGifs();
  }, [category]);





  return (
    <div>
      <h3>{category}</h3>
      {
        images?.map((img) => (
          <GifGridItem key={img.id} {...img} />
        ))
      }
    </div>
  );
};


GifGrid.propTypes = {
  category: PropTypes.string.isRequired,
};


export default GifGrid;
