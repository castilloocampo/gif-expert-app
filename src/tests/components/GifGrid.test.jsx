import { render, renderHook, waitFor } from '@testing-library/react';
import GifGrid from '../../components/Gif/GifGrid';
import useFetchGifs from '../../hooks/useFetchGifs';
import { screen } from '@testing-library/react';


describe('GifGrid component', () => {
  const category = "One Piece";
  test('should show correctly', () => {
    const { container } = render(<GifGrid category={category} />);
    expect(container).toMatchSnapshot();
  });

  test('should show gifs from category', async () => {
    const {result} = renderHook(() => useFetchGifs(category));
    render(<GifGrid category={category} />);
    await waitFor(() => {
      const elements = screen.getAllByTestId('title');
      expect(result.current.data.length).toEqual(10);
      expect(elements.length).toBe(10);
    });
    
    // expect(result.current.data).toBe(true);
  });

});
