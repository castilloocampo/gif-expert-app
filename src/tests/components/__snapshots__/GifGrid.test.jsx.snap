// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifGrid component should show correctly 1`] = `
<div>
  <h3
    class="animate__animated animate__fadeIn"
  >
    One Piece
  </h3>
  <p
    class="animate__animated animate__flash"
  >
    Loading....
  </p>
  <div
    class="card-grid"
  />
</div>
`;
